### PR TITLE
Update apiary with operator status endpoint documentation

### PR DESCRIPTION
**Pull request by ostap0207**
Related: https://github.com/salemove/backend-web/pull/1743

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -224,6 +225,47 @@ $.ajax({
 });
 ```
 
+### Events
+
+#### ENGAGEMENT_START
+
+Notifies of visitor engagement start
+
+#### ENGAGEMENT_ENT
```
Comment (deiwin) [inline apiary.apib]: typo

**Comment by deiwin** [comment]
No need to wait for backend-web deploy. This PR and the backend-web one should be merged at the same time and then when backend-web is deployed, we merge the `master` branch of this repository into the `release` branch. This will then publish the docs. Only updates to the `release` branch are now published.

**Comment by indrekj** [comment]
+1

File: apiary.apib [hunk 2]
```diff
@@ -224,6 +225,47 @@ $.ajax({
...
+
+Notifies of visitor engagement start
+
+#### ENGAGEMENT_END
+
+Notifies of visitor engagement end
+
+#### OPERATOR_STATUS_UPDATE
+
+Notifies listener of operator state updates (available, media)
```
Comment (deiwin) [inline apiary.apib]: `available` -> `availability`

File: apiary.apib [hunk 3]
```diff
@@ -224,6 +225,47 @@ $.ajax({
...
++ Parameters
+
+    + id ... Operator id
+    + name ... Operator name
+    + picture
+        + url ... Operator picture url
+    + state
+        + available ... Flag that indicates if an operator is available for a new engagement
+        + media ... List of communication types through which an operator is accessible. Possible values are:
+        'video', 'audio', 'chat'
```
Comment (deiwin) [inline apiary.apib]: use backticks over quotes for the possible values

File: apiary.apib [hunk 4]
```diff
@@ -224,6 +225,47 @@ $.ajax({
...
+  state: {
+    available: true,
+    media: ['audio', 'chat']
+  }
+}
+```
+
++ Parameters
+
+    + id ... Operator id
```
Comment (deiwin) [inline apiary.apib]: I'm unfamiliar with this kind of `...` notation. Is it also used elsewhere in the doc?

Comment (ostap0207) [inline apiary.apib]: Yeah, used in other places as well for description.

File: apiary.apib [hunk 5]
```diff
@@ -224,6 +225,47 @@ $.ajax({
 });
 ```
 
+### Events
+
+#### ENGAGEMENT_START
+
+Notifies of visitor engagement start
```
Comment (deiwin) [inline apiary.apib]: Should be something more descriptive, like: `Triggered when the visitor engages or is engaged by an operator.` Same for engagement end

Comment (deiwin) [inline apiary.apib]: As discussed the term 'engagement' will not be as familiar for people reading these docs.

Comment (ostap0207) [inline apiary.apib]: `Triggered when the visitor engages' sounds as descriptive. 
As for `engagement` it's part of our API http://docs.salemove.apiary.io/#reference/engagements, so it's ok to use it I think. Maybe there should be some general documentation or glossary though.

File: apiary.apib (hunk 4 above)
Comment (ostap0207) [inline apiary.apib]: But as I see it's mostly used by Apiary specific thing. So maybe `-` would be better here

File: apiary.apib (hunk 5 above)
Comment (deiwin) [inline apiary.apib]: Fair point on 'engagement' being described elsewhere in the documentation. Maybe we should even add a glossary to the documentation at some point. But `Triggered when the visitor engages` definitely does not read well. It feels incomplete.

File: apiary.apib (hunk 4 above)
Comment (deiwin) [inline apiary.apib]: I think it's fine for now. We could however do some cleanup and unify the usage of this. Some places don't use anything as a separator here, for example.

File: apiary.apib [hunk 6]
```diff
@@ -224,6 +225,47 @@ $.ajax({
 });
 ```
 
+### Events
+
+#### ENGAGEMENT_START
+
+Triggered when visitor engagement starts
```
Comment (deiwin) [inline apiary.apib]: The term `visitor engagement` is confusing, I think. We're free to use more verbose and descriptive language here. What did you not like about my original suggestion?

Comment (ostap0207) [inline apiary.apib]: For me `ENGAGEMENT_START` is descriptive enough. in description we should say if there are some additional input, output etc.
`engages` not sure if you can use it like this. Sound a bit weird.
`engaged by operator` , visitor can start engagement him/her self, not completely good here.
Maybe `Triggered when visitor engagement session starts` is better?

Comment (deiwin) [inline apiary.apib]: You sure can use it like this. The sentence I wrote is basically a shorthand for 'Triggered when visitor engages an operator or is engaged by an operator. ([some dict ref I found](http://www.macmillandictionary.com/us/dictionary/american/engage-in))

Comment (ostap0207) [inline apiary.apib]: Ok. But I would keep it simple. Why to make complex sentence where it's possible describe with one word.

Comment (ostap0207) [inline apiary.apib]: @deiwin are there more comments here?

Comment (deiwin) [inline apiary.apib]: Please just trust my judgement here and go with `Triggered when the visitor engages or is engaged by an operator` for the start event description and `Triggered when an engagement the visitor was in has ended` for the end one

Comment (ostap0207) [inline apiary.apib]: Ok, np

**Comment by deiwin** [comment]
!squash

**Pull request merged**